recipe/meta.yaml

```diff
@@ -13,25 +13,26 @@ source:
   patches:
     # ctypes.util.find_library() isn't CONDA_PREFIX aware :'(
     # https://github.com/ContinuumIO/anaconda-issues/issues/1716
-    - 0001-Make-load_lib-CONDA_PREFIX-aware.patch  # [unix]
+    - patches/0001-Make-load_lib-CONDA_PREFIX-aware.patch  # [unix]
 
 build:
   number: 0
-  script: '{{ PYTHON }} -m pip install . --no-deps --ignore-installed -vvv '
+  script: {{ PYTHON }} -m pip install . --no-deps --ignore-installed -vvv --no-build-isolation
 
 requirements:
   build:
     - {{ compiler('c') }}
     - patch       # [not win]
-    - m2-patch    # [win]
   host:
     - python
     - pip
     - setuptools
     - wheel
+    - gmp   # [not win]
```

Comment:
The upstream indicates that `gmp` is an optional package for speeding up public key operations in Unix systems. They mention `mpir` as an option as well (would be more beneficial on Windows if they support it). I think this makes more sense as a `run_constrained` since it does not seem to be needed at build time.

Reply:
Add `gmp` and `mpir` to `run_constrained`, or just the latter?

Reply:
@bwingconda did you see this in `run_test.py`? Conda build can't find `gmp` if it's in `run_constrained`, and tests fail without it.

https://github.com/AnacondaRecipes/pycryptodomex-feedstock/blob/584b654f7d5eec4b7fec71aa9f3148c203ce8157/recipe/run_test.py#L14

Reply:
Now I understand why it was previously in `run:` instead of `run_constrained:`. I was under the impression `gmp` was pinned to some minimum version in the upstream source. Since it is not, putting it in `run_constrained:` without a constraint doesn't do anything.

Reply:
Can you approve?